### PR TITLE
Add back lambda JSON sampling

### PR DESCRIPTION
**Pull request by raphael-theriault-swi**
Had to update the sampling logic again to actually take timestamps into account, it should be fully complete now. Also removed some star imports for code style.

### CODE REVIEW

File: packages/solarwinds-apm/src/sampling/json.ts [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+    ttl: z.number(),
+    arguments: z
+      .object({
+        BucketCapacity: z.number().optional(),
+        BucketRate: z.number().optional(),
+        TriggerRelaxedBucketCapacity: z.number().optional(),
+        TriggerRelaxedBucketRate: z.number().optional(),
+        TriggerStrictBucketCapacity: z.number().optional(),
+        TriggerStrictBucketRate: z.number().optional(),
+        SignatureKey: z.string().optional(),
```
Comment (cheempz) [inline packages/solarwinds-apm/src/sampling/json.ts:118]: minor note that otelcol will not set the SignatureKey in the json file in lambda, but no harm having it here esp. if this sampler/parser is used for the future HTTP endpoint.

**Review by cheempz — COMMENTED** [review]
Wow this was a good chance to see the pure-JS sampler, really appreciate how well organized and extensible you made it. And the tests, yay!

File: packages/solarwinds-apm/src/sampling/json.ts (hunk 1 above)
Comment (raftario) [reply packages/solarwinds-apm/src/sampling/json.ts:118]: This just reflects the parsing code in oboe

File: packages/solarwinds-apm/src/init.ts [hunk 2]
```diff
@@ -178,7 +179,22 @@ async function initTracing(
...
+
+    sampler = new JsonSampler(config, "/tmp/solarwinds-apm-settings.json")
+    processors = [
+      new TransactionNameProcessor(config),
+      new ResponseTimeProcessor(),
+      new BatchSpanProcessor(new TraceExporter(config)),
+      new ParentSpanProcessor(),
+    ]
+
+    api.waitUntilReady = () => Promise.resolve(true)
```
[the remark this answers is not in the document]
Comment (raftario) [reply packages/solarwinds-apm/src/init.ts]: Updated it to do that, it *is* a footgun though

[the remark this answers is not in the document]
Comment (cheempz) [reply packages/solarwinds-apm/src/init.ts]: yes it's debatable whether this API should be used in lambda, the one use case i can think of is scheduled invocations like cron.

**Review by cheempz — APPROVED** [review]
LGTM, thanks @raphael-theriault-swi!

**Pull request merged**